### PR TITLE
Small tweaks to get Python tests running on Windows

### PR DESCRIPTION
**Pull request by poodlewars**
There are still a few failures that I'll look at next, but this gets us as far as having most of the tests pass. It also lets us build the extension with a Python of our choice easily, and debug the extension from Python tests.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -59,14 +59,18 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-include(PythonUtils) # Must be called before Pybind (third_party) to override its finding mechanism
+if(NOT WIN32)
```
Comment (qc00) [inline cpp/CMakeLists.txt:62]: Is this still necessary given what we've learned the real cause was?

File: python/tests/conftest.py [hunk 2]
```diff
@@ -42,20 +48,16 @@
 BUCKET_ID = 0
 
 
+def run_server(port):
```
Comment (qc00) [inline python/tests/conftest.py:51]: Can we make it clear what kind of server this is?

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (poodlewars) [reply cpp/CMakeLists.txt:62]: Yes, it had issues discovering the Python static lib with this left in.

File: python/tests/conftest.py (hunk 2 above)
Comment (poodlewars) [reply python/tests/conftest.py:51]: That is fair - I'll make it private too.

**Pull request merged**